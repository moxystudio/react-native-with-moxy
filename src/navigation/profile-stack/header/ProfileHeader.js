import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import { Button, useTheme } from '../../../shared/components';

import buildStyles from './styles';

const ProfileHeader = ({ navigation }) => {
    const { themeStyles } = useTheme();
    const styles = buildStyles(themeStyles);

    const handlePress = useCallback(() => {
        navigation.pop();
    }, [navigation]);

    return (
        <View style={ styles.header }>
            <Button title="Go back" textStyle={ styles.goBack } onPress={ handlePress } />
            <Text style={ styles.title }>Profile Stack</Text>
        </View>
    );
};

ProfileHeader.propTypes = {
    navigation: PropTypes.object.isRequired,
};

export default ProfileHeader;
