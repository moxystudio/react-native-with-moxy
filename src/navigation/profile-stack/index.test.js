import React from 'react';
import ProfileStack from './';
import { shallow } from 'enzyme';
import ProfileHeader from './header';

it('should create navigation correctly', () => {
    const tree = shallow(<ProfileStack />);
    const StackNavigator = tree.find('StackNavigator').first();

    expect(StackNavigator.props().screenOptions.header).toEqual(ProfileHeader);

    expect(tree).toMatchSnapshot();
});
