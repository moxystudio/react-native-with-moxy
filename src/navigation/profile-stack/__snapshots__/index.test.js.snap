// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create navigation correctly 1`] = `
<StackNavigator
  screenOptions={
    Object {
      "header": [Function],
    }
  }
>
  <Screen
    component={[Function]}
    name="Profile1"
  />
  <Screen
    component={[Function]}
    name="Profile2"
  />
</StackNavigator>
`;
